src/modules/Builder/features/ActivityItemsFlow/ActivityItemsFlow.tsx

```diff
@@ -46,10 +53,17 @@ export const ActivityItemsFlow = () => {
     handleCloseRemovePopup();
   };
 
+  const { data, isPending } = useDataPreloader<ConditionalLogic>({
+    data: flowItems,
+    rootSelector: `.${ACTIVITY_ITEMS_FLOW_LIST_CLASS}`,
+    targetSelector: `.${ACTIVITY_ITEMS_FLOW_END_ITEM_CLASS}`,
+  });
+
   const headerProps = {
-    isAddItemFlowDisabled: items?.length < 2,
+    isAddItemFlowDisabled: items?.length < 2 || isPending,
     onAddItemFlow: handleAddItemFlow,
   };
+  const contentSx = { minHeight: '15rem', gap: '2.4rem', position: 'relative' };
```

Comment:
can be moved to ActivityItemsFlow.const.ts

Reply:
done